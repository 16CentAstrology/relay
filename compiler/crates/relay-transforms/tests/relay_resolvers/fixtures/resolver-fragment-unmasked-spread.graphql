fragment Foo_user on User {
  pop_star_name
  # it should be fine to use this fragment spread here
  ...UnmaskedFragment
}

fragment PopStarNameResolverFragment_name on User {
  name
  # This should be allowed since UnmaskedFragment has @relay(mask: false)
  ...UnmaskedFragment @relay(mask: false)
}

fragment UnmaskedFragment on User {
  address {
    street
  }
}

# %extensions%

extend type User {
  pop_star_name: String @relay_resolver(fragment_name: "PopStarNameResolverFragment_name", import_path: "PopStarNameResolver")
}