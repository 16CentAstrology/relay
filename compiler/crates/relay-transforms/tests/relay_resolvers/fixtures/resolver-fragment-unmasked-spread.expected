==================================== INPUT ====================================
fragment Foo_user on User {
  pop_star_name
  # it should be fine to use this fragment spread here
  ...UnmaskedFragment
}

fragment PopStarNameResolverFragment_name on User {
  name
  # This should be allowed since UnmaskedFragment has @relay(mask: false)
  ...UnmaskedFragment @relay(mask: false)
}

fragment UnmaskedFragment on User {
  address {
    street
  }
}

# %extensions%

extend type User {
  pop_star_name: String @relay_resolver(fragment_name: "PopStarNameResolverFragment_name", import_path: "PopStarNameResolver")
}
==================================== OUTPUT ===================================
fragment Foo_user on User {
  ...PopStarNameResolverFragment_name @__RelayResolverMetadata
  # RelayResolverMetadata {
  #     field_id: FieldID(526),
  #     import_path: "PopStarNameResolver",
  #     import_name: None,
  #     field_alias: None,
  #     field_path: "pop_star_name",
  #     field_arguments: [],
  #     live: false,
  #     output_type_info: Legacy,
  #     fragment_data_injection_mode: None,
  #     type_confirmed: false,
  #     resolver_type: ResolverModule,
  # }
  
  ...UnmaskedFragment
}

fragment PopStarNameResolverFragment_name on User {
  name
  ...UnmaskedFragment @relay(mask: false)
}

fragment UnmaskedFragment on User {
  address {
    street
  }
}
